### PR TITLE
chore: Adds release workflows

### PR DESCRIPTION
**Pull request by yorik**
# What ❔

Adds staging release workflows.
Adds post-merge prometheus statistic.
Moves from unsupported actions-rs/toolchain to actions-rust-lang/setup-rust-toolchain in ci-prover-reusable.yml.
Adds devops as CODEOWNERS for Github workflows

## Why ❔

To create staging releases from this repo.

## Checklist

- [x] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.

### CODE REVIEW

**Review by shahar4 — APPROVED** [review]

**Comment by andreyg-matterlabs** [comment]
Looks good

**Review by shahar4 — APPROVED** [review]

File: .github/workflows/build-gar-reusable.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        type: boolean
+
+jobs:
+  build-gar-prover:
+    name: Build GAR prover
+    runs-on: [self-hosted, ci-runner]
+    strategy:
+      fail-fast: false
+      matrix:
+        # https://github.com/matter-labs/gitops-kubernetes/blob/main/apps/base/zksync-v2-config/zksync-v2-config.yaml#L385
```
Comment (Deniallugo) [inline .github/workflows/build-gar-reusable.yml:27]: should we expose it?

**Review by Deniallugo — APPROVED** [review]
just a small question

File: .github/workflows/build-gar-reusable.yml (hunk 1 above)
Comment (yorik) [reply .github/workflows/build-gar-reusable.yml]: Removed, I hope we have this knowledge somewhere.

**Review by Deniallugo — APPROVED** [review]

**Pull request merged**